torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -4936,6 +4931,78 @@ def _test_post_localSGD_optimizer_parity(self, create_averager, grad_is_view):
         def _create_periodic_model_averager(self):
             return averagers.PeriodicModelAverager(period=4, warmup_steps=10)
 
+        def _create_post_localSGD_optimizer(self, net, learning_rate, averager):
+            return post_localSGD_optimizer.PostLocalSGDOptimizer(
+                optim=torch.optim.SGD(net.parameters(), lr=learning_rate),
+                averager=averager,
+            )
+
+        def _perform_a_train_step(self, optimizer, net, loss_fn, input, target):
+            optimizer.zero_grad()
+            output = net(input)
+            loss = loss_fn(output, target)
+            loss.backward()
+            optimizer.step()
+
+        def _test_post_localSGD_optimizer_step_reload(self, create_averager):
+            learning_rate = 0.03
+            chkpt_file = tempfile.gettempdir() + "/checkpoint.pt"
+
+            net_using_post_localSGD_opt = torch.nn.parallel.DistributedDataParallel(
+                copy.deepcopy(DDP_NET).cuda(),
+                device_ids=[self.rank]
+            )
+
+            averager = create_averager()
+            post_localSGD_opt = self._create_post_localSGD_optimizer(net_using_post_localSGD_opt,
+                                                                     learning_rate,
+                                                                     averager)
+
+            averager2 = create_averager()
+            dummy_post_localSGD_opt = self._create_post_localSGD_optimizer(net_using_post_localSGD_opt,
+                                                                           learning_rate,
+                                                                           averager2)
+
+            input = torch.randn(dist.get_world_size() * 2, 2).cuda()
+            target = torch.randn(dist.get_world_size() * 2, 4).cuda()
+            loss_fn = nn.MSELoss()
+
+            for _ in range(20):
+                self._perform_a_train_step(
+                    post_localSGD_opt,
+                    net_using_post_localSGD_opt,
+                    loss_fn,
+                    input,
+                    target
+                )
+
+            if self.rank == 0:
+                torch.save({'optimizer_state_dict': post_localSGD_opt.state_dict()}, chkpt_file)
+
+            dist.barrier()
+            map_location = {'cuda:%d' % 0: 'cuda:%d' % self.rank}
+            checkpoint = torch.load(chkpt_file, map_location=map_location)
+            dummy_post_localSGD_opt.load_state_dict(checkpoint['optimizer_state_dict'])
+
+            # Check if dummy averager was initialized to a correct value
+            self.assertIsNot(averager2.step, 0)
```

Comment:
We would want to check `averager2.step != 0` instead of `not averager2.step is 0` since `is` checks for `int`s in Python may not be reliable: https://stackoverflow.com/questions/2239737/is-it-better-to-use-is-or-for-number-comparison-in-python

I think the change would be:

```suggestion
            self.assertNotEqual(averager2.step, 0)
```

However, should we just check the `.step` value to the desired positive integer (i.e. 20)? Otherwise, a passing implementation could simply set `step` to `1` if there is an entry in the state dict and `0` otherwise.

Reply:
Valid point. Addressed it.